velox/exec/RowsStreamingWindowBuild.cpp

```diff
@@ -102,12 +123,22 @@ void RowsStreamingWindowBuild::noMoreInput() {
 
 std::shared_ptr<WindowPartition> RowsStreamingWindowBuild::nextPartition() {
   VELOX_CHECK(hasNextPartition());
-  return windowPartitions_[++outputPartition_];
+  auto output = windowPartitions_.front();
+  outputPartition_++;
+  // If the output partition index is less than the input partition index,
+  // it means the current output partition has been processed and there are new
+  // input partitions. Therefore, we need to remove the processed output
+  // partition from the queue.
+  if (outputPartition_ < inputPartition_) {
```

Comment:
This is a bit counter-intuitive to understand... This code should generally pop the windowPartitions_ except when outputPartition_ == inputPartition_.

Do we ever encounter the case when outputPartition_ > inputPartition_ ?

Reply:
@aditi-pandit  The outputPartition_ is always less than or equal to inputPartition_ and will never exceed it. It seems we can remove the code in addInput that handles the removal of processed partitions and keep this logic only in nextPartition(). cc @FelixYBW

Reply:
If we remove the pop from addPartitionInputs, the outputPartition_ may be output twice, will it lead to error?

```
    if (outputPartition_ == inputPartition_) {
      // If the output partition is the same as the input partition, it has
      // already been output,
      // so we need to remove the output partition from the queue.
      windowPartitions_.pop_front();
    }
```